src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1582,48 +1580,33 @@ private void PerformPredicateAnalysisCore(IOperation operation, TAnalysisData ta
-                                                equals: FlowBranchConditionKind == ControlFlowConditionKind.WhenFalse, isReferenceEquality: false, targetAnalysisData: targetAnalysisData);
-                                        }
-
-                                        break;
-                                    }
-                                    else
-                                    {
-                                        goto default;
-                                    }
+                                predicateValueKind = SetValueForEqualsOrNotEqualsComparisonOperator(isPatternOperation.Value, negatedConstantPattern.Value,
+                                    equals: FlowBranchConditionKind == ControlFlowConditionKind.WhenFalse, isReferenceEquality: false, targetAnalysisData: targetAnalysisData);
+                            }
 
-                                case "RelationalPattern":
-                                    // For the true branch, set the pattern operation value to NotNull.
-                                    if (FlowBranchConditionKind == ControlFlowConditionKind.WhenTrue)
-                                    {
-                                        predicateValueKind = SetValueForIsNullComparisonOperator(isPatternOperation.Value, equals: false, targetAnalysisData: targetAnalysisData);
-                                    }
+                            break;
 
-                                    break;
+                        case OperationKind.RelationalPattern:
+                            // For the true branch, set the pattern operation value to NotNull.
+                            if (FlowBranchConditionKind == ControlFlowConditionKind.WhenTrue)
+                            {
+                                predicateValueKind = SetValueForIsNullComparisonOperator(isPatternOperation.Value, equals: false, targetAnalysisData: targetAnalysisData);
+                            }
 
-                                case "BinaryPattern":
-                                    // These high level patterns should not be present in the lowered CFG: https://github.com/dotnet/roslyn/issues/47068
-                                    predicateValueKind = PredicateValueKind.Unknown;
-                                    break;
+                            break;
 
-                                default:
-                                    Debug.Fail($"Unknown pattern kind '{isPatternOperation.Pattern.Kind}'");
-                                    predicateValueKind = PredicateValueKind.Unknown;
-                                    break;
-                            }
+                        case OperationKind.BinaryPattern:
+                            // These high level patterns should not be present in the lowered CFG: https://github.com/dotnet/roslyn/issues/47068
```

Comment:
This is the root cause for #6755. We will need to figure out performing flow analysis in presence of binary pattern operations. I am working on a separate PR for it.

Reply:
It could be for a bunch of CA1508 too. I'd love to see the correct handling of this will be!